app/baseapp.go

```diff
@@ -26,20 +28,20 @@ type Application interface {
 	Query(req abci.RequestQuery) abci.ResponseQuery
 
 	MakeCacheContext(header tmproto.Header) (ctx sdk.Context, writer func())
-	AppCodec() codec.Marshaler
+	AppCodec() codec.Codec
 	GetIBCKeeper() ibckeeper.Keeper
 }
 
 type BaseApp struct {
 	logger           log.Logger
-	name             string            // application name from abci.Info
-	db               dbm.DB            // common DB backend
-	cms              *store.Store      // Main (uncached) state
-	router           sdk.Router        // handle any kind of message
-	queryRouter      sdk.QueryRouter   // router for redirecting query calls
-	grpcQueryRouter  *GRPCQueryRouter  // router for redirecting gRPC query calls
-	msgServiceRouter *MsgServiceRouter // router for redirecting Msg service messages
-	txDecoder        sdk.TxDecoder     // unmarshal []byte into sdk.Tx
+	name             string                    // application name from abci.Info
+	db               dbm.DB                    // common DB backend
+	cms              *store.Store              // Main (uncached) state
+	router           sdk.Router                // handle any kind of message
+	queryRouter      sdk.QueryRouter           // router for redirecting query calls
+	grpcQueryRouter  *GRPCQueryRouter          // router for redirecting gRPC query calls
```

Comment:
`GRPCQueryRouter` (and `app/grpcrouter.go`) looks the same as cosmos/cosmos-sdk at a quick glance, is there any reason why we are using our own definition?

Reply:
This is because `GRPCQueryRouter.returnTypeOf` is private method.